src/dodal/parameters/gda_directory_provider.py

```diff
@@ -0,0 +1,71 @@
+from functools import lru_cache
+from typing import Optional
+
+from aiohttp import ClientSession
+from ophyd_async.core import DirectoryInfo, DirectoryProvider
+from pydantic import BaseModel
+
+
+class DataCollectionIdentifier(BaseModel):
+    collectionId: int
+
+
+class VisitDirectoryProviderConfig(BaseModel, frozen=True):
+    url: str
+    beamline: str
+    base_path: str
```

Comment:
VDPC then becomes part of BlueAPI config and is passed down as part of make_directory_provider when constructing the singleton directory provider.